src/main/kotlin/org/rust/lang/core/psi/ext/RsAttr.kt

```diff
@@ -5,8 +5,18 @@
 
 package org.rust.lang.core.psi.ext
 
+import org.rust.lang.core.psi.RsInnerAttr
 import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsOuterAttr
 
 interface RsAttr : RsElement {
     val metaItem: RsMetaItem
 }
+
+val RsAttr.owner: RsDocAndAttributeOwner?
+    get() = when (this) {
+        is RsInnerAttr -> parent?.parent as? RsDocAndAttributeOwner
+        is RsOuterAttr -> parent as? RsDocAndAttributeOwner
+        // Throw exception so that any problems with this property aren't silent and are instead easily findable
+        else -> throw IllegalArgumentException("Unsupported attribute type: $this")
```

Comment:
You can use [error](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/error.html) here. It throws `IllegalStateException`, but I think it makes more sense, because now `else` branch is unreachable.